plugins/pluginmanager.go

```diff
@@ -41,26 +41,24 @@ func mergeErrors(errs ...error) error {
 	return nil
 }
 
-// waitForErrors returns when it has received count results from the errors
-// channel.
+// waitForErrors executes function f against every plugin passed in taskPlugins
+// and returns an error which represents the merge of all errors which occurred
+// against any plugin.
 //
 // Note, that errors might be nil, if all are nil it'll return nil otherwise
 // it'll merge the errors.
-func waitForErrors(errors <-chan error, count int) error {
-	if count == 0 {
-		return nil
+func waitForErrors(taskPlugins []TaskPlugin, f func(p TaskPlugin) error) (err error) {
+	errors := make([]error, len(taskPlugins))
+	var wg sync.WaitGroup
+	for i, j := range taskPlugins {
+		wg.Add(1)
+		go func(i int, j TaskPlugin) {
```

Comment:
yeah, golang doesn't capture variables from `for` loops properly... this is like javascript without `let`